src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryServicesTests.cs

```diff
@@ -0,0 +1,147 @@
+            {
+                string ouName = "NetCoreDevs";
+                CreateOU(de, ouName, ".Net Core Developers Unit");
+                try
+                {
+                    SearchOUByName(de, ouName);
+                }
+                finally 
+                {
+                    // Clean up the added ou
+                    DeleteOU(de, ouName);
+                }
+            }
+        }
+
+        [ConditionalFact(nameof(IsLdapConfigurationExist))]
+        public void TestCN() // adding and removing organization unit
+        {
+			using (DirectoryEntry de = new DirectoryEntry(LdapConfiguration.Configuration.LdapPath,
+														    LdapConfiguration.Configuration.UserName,
+														    LdapConfiguration.Configuration.Password,
+														    LdapConfiguration.Configuration.AuthenticationTypes))
+            {
+                DirectoryEntry rootOU = CreateOU(de, "CoreFxRootOU", "CoreFx Root OU");
+                try
+                {
+                    DirectoryEntry child1OU = CreateOU(rootOU, "CoreFxChild1OU", "CoreFx Child OU 1");
+                    DirectoryEntry child2OU = CreateOU(rootOU, "CoreFxChild2OU", "CoreFx Child OU 2");
+
+                    CreateCN(child1OU, "user.ou1.1", "User 1 is in CoreFx ou 1", "1 111 111 11111");
+                    CreateCN(child1OU, "user.ou1.2", "User 2 is in CoreFx ou 1", "1 222 222 22222");
+
+                    CreateCN(child2OU, "user.ou2.1", "User 1 is in CoreFx ou 2", "1 333 333 3333");
+                    CreateCN(child2OU, "user.ou2.2", "User 2 is in CoreFx ou 2", "1 333 333 3333");
+
+                    // now let's search for the added data:
+                    SearchOUByName(rootOU, "CoreFxChild1OU");
+                    SearchOUByName(rootOU, "CoreFxChild2OU");
+
+                    SearchCNByName(child1OU, "user.ou1.1");
+                    SearchCNByName(child1OU, "user.ou1.2");
+
+                    SearchCNByName(child2OU, "user.ou2.1");
+                    SearchCNByName(child2OU, "user.ou2.2");
+                }
+                finally
+                {
+                    // rootOU.DeleteTree(); doesn't work as getting "A protocol error occurred. (Exception from HRESULT: 0x80072021)"
+                    DeleteOU(de, "CoreFxRootOU");
+                }
+            }
+        }
+
+        private DirectoryEntry CreateOU(DirectoryEntry de, string ou, string description)
+        {
+            DirectoryEntry ouCoreDevs = de.Children.Add($"ou={ou}","Class");
+            ouCoreDevs.Properties["objectClass"].Value = "organizationalUnit";    // has to be organizationalUnit
+            ouCoreDevs.Properties["description"].Value = description;
+            ouCoreDevs.Properties["ou"].Value = ou;
+            ouCoreDevs.CommitChanges();
+            return ouCoreDevs;
+        }
+
+        private DirectoryEntry CreateCN(DirectoryEntry ouEntry, string cn, string description, string phone)
```

Comment:
CN isn't really a "type" it stands for common name.  I would change this name to match the object type that is actually getting created.

Reply:
Thanks. I'll fix that